### ISSUE TITLE
'by_chanels' normalization issue between different splits

### ISSUE THREAD

**Issue by sibosutd** [ikhlestov/vision_networks#27]
Hi Illarion,

Thanks for the beautiful code. 

I have a question regarding the 'by_chanels' normalization on the CIFAR-10 test set. It seems like, when preprocessing the CIFAR-10 test set, you are computing the means of test set instead of training set since the CifarDataSet objects compute the individual mean of different splits(train/val/test). However, would it be correct that the preprocessing statistics has to come only from the training set?
(Reference: http://cs231n.github.io/neural-networks-2/, 'common pitfalls' paragraph)
Correct me if I misread your code.

Again, thanks for your work and It is a great implementation of DenseNet.

**Comment by ikhlestov** [comment]
Hi,

Yes, you must be right. I'll try to fix it, but I don't know exactly when I have free time.

Thank you for reporting the issue.

**Comment by sibosutd** [comment]
@ikhlestov Thanks for your reply.

It's just several lines of code. I can update the code with a PR if you don't mind.

**Comment by ikhlestov** [comment]
It will be cool. I'll try to test it on some of my servers.
